### PR TITLE
add mobile availability grid

### PR DESCRIPTION
**Pull request by learykara**
Adds a mobile availability grid.  Right now this only detects browser width on initial page load, so dragging and resizing your screen won't do anything.

image

### CODE REVIEW

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ const moment = require('moment');
 const chunk = require('lodash/array/chunk');
 const difference = require('lodash/array/difference');
 const fill = require('lodash/array/fill');
+const { Icon } = require('../Icon');
```
Comment (ripleyaffect) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: Should be in a different group

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 2]
```diff
@@ -17,7 +18,8 @@ const AvailabilityGridSlot = React.createClass({
     data: React.PropTypes.object,
     onSelectionModeChanged: React.PropTypes.func,
     onSlotUnselected: React.PropTypes.func,
-    onSlotSelected: React.PropTypes.func
+    onSlotSelected: React.PropTypes.func,
```
Comment (ripleyaffect) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: any `isRequired`s here?

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 3]
```diff
@@ -69,7 +72,10 @@ const AvailabilityGridDay = React.createClass({
   propTypes: {
     minSlot: React.PropTypes.number,
     maxSlot: React.PropTypes.number,
-    data: React.PropTypes.object
+    data: React.PropTypes.object,
+    onNavigateDay: React.PropTypes.func,
+    isMinDay: React.PropTypes.bool,
+    isMaxDay: React.PropTypes.bool,
```
Comment (ripleyaffect) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: any `isRequired`s required?

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 4]
```diff
@@ -84,9 +90,29 @@ const AvailabilityGridDay = React.createClass({
     })
 
     return (
-      <div className="availability-grid-day">
-        {this.props.data.name}
-        {slotNodes.slice(this.props.minSlot, this.props.maxSlot)}
+      <div className={classNames(
+          "availability-grid-day",
+          {"availability-grid-day__mobile": this.props.mobile})}>
```
Comment (ripleyaffect) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: why does this have to be done via `props` and not media query?

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 5]
```diff
@@ -84,9 +90,29 @@ const AvailabilityGridDay = React.createClass({
...
+          {this.props.mobile &&
+            <Icon
+                className={classNames(
+                  "navigation navigation__right",
+                  {disabled: this.props.isMaxDay})}
+                name="navigateright"
+                onClick={e => this.props.onNavigateDay(1)}/>}
+        </span>
+        <div className="availability-grid-items">
+          {slotNodes.slice(this.props.minSlot, this.props.maxSlot)}
```
Comment (ripleyaffect) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: the `maxSlot` is exclusive here, should it be `maxSlot + 1`?

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 6]
```diff
@@ -109,7 +136,15 @@ const AvailabilityGrid = React.createClass({
...
   getInitialState() {
-    const days = ['Mon', 'Tues', 'Wed', 'Thurs', 'Fri', 'Sat', 'Sun'];
+    let days;
+    if (this.shouldRenderMobile()) {
+      days = [
+        'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'
+      ];
+    }
+    else {
+      days = ['Mon', 'Tues', 'Wed', 'Thurs', 'Fri', 'Sat', 'Sun'];
```
Comment (ripleyaffect) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: why are `Tues` and `Thurs` 4 letters and the rest 3? they're just as identifiable w/ 3 letters and this is preventing you from generating this with a map that cuts down to the first three letters

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 7]
```diff
@@ -302,6 +338,16 @@ const AvailabilityGrid = React.createClass({
     }
   },
 
+  shouldRenderMobile() {
+    return window.innerWidth < 620;
```
Comment (ripleyaffect) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: magic number alert!

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 8]
```diff
@@ -322,8 +376,12 @@ const AvailabilityGrid = React.createClass({
             onSelectionModeChanged={this.handleSelectionModeChanged}
             onSlotSelected={this.handleSlotSelected}
             onSlotUnselected={this.handleSlotUnselected}
+            onNavigateDay={this.handleNavigateDay}
```
Comment (ripleyaffect) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: not in alphabetical order :scream:

File: src/AvailabilityGrid/AvailabilityGrid.jsx [hunk 9]
```diff
@@ -339,10 +397,12 @@ const AvailabilityGrid = React.createClass({
...
           onMouseDown={this.onMouseDown}
           onMouseUp={this.onMouseUp} >
-        <div
-            className="availability-grid-slot-names">
-          {slotNames.slice(minSlot, maxSlot)}
-        </div>
+        {!this.shouldRenderMobile() &&
+          <div
+              className="availability-grid-slot-names">
+            {slotNames.slice(minSlot, maxSlot)}
```
Comment (ripleyaffect) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: ditto on exclusive thing

File: src/AvatarUploader/AvatarUploader.jsx [hunk 10]
```diff
@@ -86,6 +86,11 @@ class AvatarUploader extends React.Component {
     }, this.props.onChange);
   }
 
+  _isOnMobile = () => {
+    // Smart phones have an orientation property. Use that to detect mobile
```
Comment (ripleyaffect) [inline src/AvatarUploader/AvatarUploader.jsx]: huh? what about tablets?

**Comment by ripleyaffect** [comment]
:+1: apt from comments

File: src/AvailabilityGrid/AvailabilityGrid.jsx (hunk 5 above)
Comment (learykara) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: hmm.. not sure. existing code and I believe it's all working the way we want

File: src/AvailabilityGrid/AvailabilityGrid.jsx (hunk 6 above)
Comment (learykara) [inline src/AvailabilityGrid/AvailabilityGrid.jsx]: :shrug: wasn't me. no idea if intentional or not

File: src/AvatarUploader/AvatarUploader.jsx (hunk 10 above)
Comment (learykara) [inline src/AvatarUploader/AvatarUploader.jsx]: can't upload an image from tablets either, yet

**Pull request merged**